pkg/ingester/instance.go

```diff
@@ -588,9 +588,28 @@ type UniqueValues map[string]struct{}
 
 // LabelsWithValues returns the label names with all the unique values depending on the request
 func (i *instance) LabelsWithValues(ctx context.Context, startTime time.Time, matchers ...*labels.Matcher) (map[string]UniqueValues, error) {
-	// TODO (shantanu): Figure out how to get the label names from index directly when no matchers are given.
-
 	labelMap := make(map[string]UniqueValues)
+	if len(matchers) == 0 {
+		labelsFromIndex, err := i.index.LabelNames(startTime, nil)
```

Comment:
yeah, I noticed there wasn't any instance where we're invoking the function with a non-nil shard value.. my assumption was it's because we're querying the index here?

Reply:
got it ... our interfaces are getting very overloaded :(